commands/db.bee.inc

```diff
@@ -120,6 +120,26 @@ function sql_bee_callback($arguments, $options) {
   $db_info = $db_connection['default'];
 
   // Open SQL command-line.
-  $command = 'mysql --user=' . $db_info['username'] . ' --password=' . $db_info['password'] . ' --host=' . $db_info['host'] . ' ' . $db_info['database'];
+  $command = 'mysql ' . db_bee_mysql_options($db_info);
   proc_close(proc_open($command, array(STDIN, STDOUT, STDERR), $pipes));
 }
+
+/**
+ * Returns a decoded options string for use with MySQL commands.
+ *
+ * @param array $db_info
+ *   Connection information for the current database. Generally obtained via
+ *   Database::getConnectionInfo().
+ *
+ * @return string
+ *   A string of command options that can be appended to 'mysql' or 'mysqldump'.
+ */
+function db_bee_mysql_options($db_info) {
+
+  $options = ' --user=' . rawurldecode($db_info['username']);
```

Comment:
Please remove this leading blank line and the leading space before the `--user` (that's already accounted for when we call `db_bee_mysql_options()` above).

```suggestion
  $options = '--user=' . rawurldecode($db_info['username']);
```

Reply:
Done